### DISCUSSION TITLE
How broad is the "Read access to members and metadata" in the context of a single repo in a Github organization?

### DISCUSSION THREAD

**Discussion by b0tting** [community/community#126248] [Repositories] [answered]
### Select Topic Area

Question

### Body

We use a Github organization to manage repositories for project for our students. Students sometimes want to give external application access to their repositories, which end up as a permission request in our organization. My question is regarding the "Read access to members and metadata" permission these apps often require. 

In the request screen I can give an app permissions to either all repos in my organization, or to a single repo. If I give the app access to a single repository, will it still be able to see the details of all of our organization members? Including the 99% of members who do not have a role in that specific repository? 

Looking at the Github API, I see a "read member data" endpoint at the organization level and not on the repository level, which seems to indicate any app with read permission to a single repository can still see all organization members.

**Comment by mranv** [comment, marked as answer]
When an application requests "Read access to members and metadata" permission for your GitHub organization, it means that the application will be able to access information about the members and metadata of the entire organization, regardless of whether the permission is granted for all repositories or a single repository.

The "Read member data" endpoint you mentioned is an organization-level API endpoint, which means that if an application has access to any repository within the organization, it can access the organization's member data, including the list of members and their respective roles.

Therefore, even if you grant the application access to a single repository, it will still be able to see the details of all members of your organization, including the members who do not have a role in that specific repository.

This behavior is by design and is intended to provide a consistent and unified view of the organization's structure and membership, regardless of the specific repository access granted to an application.

If you want to limit an application's access to only the members who have roles in a specific repository, you would need to implement additional checks and filtering within your application logic, as the GitHub API itself does not provide this level of granular access control.

**Reply by b0tting** [reply]
That's a fast and clear, well written answer. Thank you.
